### PR TITLE
Enable .NET analyzers and Upgrade langversion

### PR DESCRIPTION
**Pull request by singhashish-wpf**
Enable .NET analyzers and Upgrade langversion

This is an experimental effort, NOT to be merged. 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9237)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Consider using https://github.com/dotnet/winforms/blob/main/.editorconfig as a starting point for what you enable/disable. You can drop multiple ones in your tree as you work through things, just remove `root = true` lower in the tree. Feel free to ping me on rationale for any rules.

**Comment by ThomasGoulet73** [comment]
Should this PR use the existing [infrastructure](https://github.com/dotnet/wpf/blob/fb350a6c36a9d15c75f9c7a76e0b3616201ebc7f/eng/WpfArcadeSdk/tools/CodeAnalysis.targets) for code analysis by setting `EnableAnalyzers` to true instead of doing it manually in the project ? This would use the same analyzers as System.Xaml (The only project using it right now AFAIK). There is also an already configured [config file](https://github.com/dotnet/wpf/blob/fb350a6c36a9d15c75f9c7a76e0b3616201ebc7f/eng/WpfArcadeSdk/tools/CodeAnalysis/CodeAnalysis.src.globalconfig) for the rules (Which I copied from WinForms).

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/PresentationFramework.csproj [hunk 1]
```diff
@@ -9,12 +9,23 @@
     <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
     <Platforms>AnyCPU;x64;arm64</Platforms>
     <GenerateDependencyFile>false</GenerateDependencyFile>
+    <LangVersion>latest</LangVersion>
```
Comment (sharwell) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/PresentationFramework.csproj:12]: 💡 `latest` is a moving target and not a real version. Would recommend using a numeric value for build stability across users who work on the project.

File: .editorconfig [hunk 2]
```diff
@@ -33,26 +33,32 @@ csharp_indent_braces = false
...
 csharp_indent_switch_labels = true
 csharp_indent_labels = one_less_than_current
+[src/**/*.cs]
+file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.
+
+# C# and VB files
+[*.{cs,vb}]
+charset = utf-8-bom
+insert_final_newline = true
+trim_trailing_whitespace = true
```
[the remark this answers is not in the document]
Comment (miloush) [reply .editorconfig:43]: Can you give an example?

[the remark this answers is not in the document]
Comment (sharwell) [reply .editorconfig:43]: Trailing spaces in a verbatim string would be one example. The C# formatter removes _non-semantic_ trailing whitespaces automatically. The only difference that should exist between `trim_trailing_whitespace` and the formatter is `trim_trailing_whitespace` also removes trailing whitespaces that change the meaning of code. Other differences would be considered formatter bugs that Roslyn intends to fix.

**Comment by singhashish-wpf** [comment]
Closing this in favor of using the existing infrastructure.

**Pull request closed without merging**